README.md

```diff
@@ -83,9 +83,86 @@ It should thus be useful to make sure that your python environment is set up cor
 
 ## Pre-processing
 An overview of how the different scripts and files depend on each other is given in this figure:
-<p align="middle">
-  <img src="figures/component_dependencies.png"/>
-</p>
+
+```mermaid
+graph LR
+
+%% styling
+classDef proc_step stroke:#ffa49b
+classDef npy_file stroke:#8cdecf
+classDef pt_file stroke:#8fd7f1
+
+%% processing steps
+create_grid_features_step["create_grid_features.py"]:::proc_step
+create_mesh_step["create_mesh.py"]:::proc_step
+create_parameter_weights_step["create_parameter_weights.py"]:::proc_step
+
+%% data-collections
+
+%% files
+source_xy_coordinates["nwp_xy.npy"]:::npy_file
+surface_geopotential["surface_geopotential.npy"]:::npy_file
+grid_features["grid_features.pt"]:::pt_file
+border_mask["border_mask.npy"]:::npy_file
+nwp_mbr_files["nwp_*_mbr*.npy: [Nt,Ny,Nx,Nv]"]:::npy_file
+wtr_files["wtr_*.npy: [Ny,Nx]"]:::npy_file
+nwp_toa_files["nwp_toa*.npy: [Nt,Ny,Nx]"]:::npy_file
+
+border_mask -- read by --> training_step
+grid_features -- read by --> training_step
+parameter_weights -- read by --> training_step
+mesh_information -- read by --> training_step
+dynamic_variables -- read by --> training_step
+
+subgraph Dataset
+  subgraph static_variables["static variables [Nx,Ny]"]
+    source_xy_coordinates
+    surface_geopotential
+    border_mask
+    note("stored in `data/{dataset_name}/static/")
```

Comment:
How tricky would it be to make these notes look more different than the files? It would be nice just to avoid confusion. But if it is not trivial to change the styling a bit it doesn't matter.